test/stat/test_stat.c

```diff
@@ -193,6 +193,51 @@ void test() {
   err = stat("folder/subdir", &s);
   assert(s.st_mtime != TEST_TIME);
 
+  EM_ASM(
+    var stats = FS.stat("folder/file");
+    assert(stats.dev == 1);
+    assert(stats.ino);
+    assert(stats.mode == 33279);
```

Comment:
```suggestion
    assert(stats.mode == 0100777);
```
The permissions bits are clearer this way, and the rest seems to as well..?

Reply:
I agree. One small thing is that I had to use `0o100777` in the JavaScript because it said that octal literals were not allowed in strict mode, even though I remember that we should try to keep the octal notation consistent. Are there any ways to use consistent notation while keeping strict mode on?

Reply:
Oh, interesting... I didn't know JS had a preference for the `0o` form, in strict mode. Well, in that case I guess let's do what JS prefers in JS, even though it's not consistent with C.